src/network/networkd-json.c

```diff
@@ -1050,6 +1051,79 @@ static int dhcp6_client_append_json(Link *link, JsonVariant **v) {
         return json_append_one(v, "DHCPv6Client", w);
 }
 
+static int dhcp4_client_pd_append_json(Link *link, JsonVariant **v) {
+        _cleanup_(json_variant_unrefp) JsonVariant *addresses = NULL, *array = NULL;
+        uint8_t ipv4masklen;
+        uint8_t sixrd_prefixlen;
+        struct in6_addr sixrd_prefix;
+        const struct in_addr *br_addresses;
+        size_t n_br_addresses = 0;
+        size_t n = 0;
+        int r;
+
+        assert(link);
+        assert(v);
+
+        if (!link->dhcp_lease || !dhcp4_lease_has_pd_prefix(link->dhcp_lease))
+                return 0;
+
+        r = sd_dhcp_lease_get_6rd(link->dhcp_lease, &ipv4masklen, &sixrd_prefixlen, &sixrd_prefix, &br_addresses, &n_br_addresses);
+        if (r < 0)
+                return r;
+
+        for (n = 0; n < n_br_addresses; ++n) {
+                _cleanup_(json_variant_unrefp) JsonVariant *e = NULL;
+                r = json_build(&e, JSON_BUILD_IN4_ADDR(br_addresses + n));
+                if (r < 0)
+                        return r;
+
+                r = json_variant_append_array(&addresses, e);
+        }
+
+        r = json_build(&array, JSON_BUILD_OBJECT(
+                                       JSON_BUILD_PAIR_IN6_ADDR("Prefix", &sixrd_prefix),
+                                       JSON_BUILD_PAIR_INTEGER("PrefixLength", sixrd_prefixlen),
```

Comment:
Here and below, `_INTEGER` -> `_UNSIGNED` ?

Reply:
Done.